### ISSUE TITLE
add overload for CharSequence.containsRegex which expects Regex

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#160]
*Platform* (jvm, jdk8, js, android): all

## Code related feature
```
expect("hello").containsRegex(Regex("(h|H)ello"))
expect("hello").contains.atLeast(1).regex(Regex("(h|H)ello"))

//in addition to

expect("hello").containsRegex("(h|H)ello")
expect("hello").contains.atLeast(1).regex("(h|H)ello")
```

Following the things you need to do:

*domain*
- modify CharSequenceContainsAssertions, expect a `List<Regex>` instead of a `List<String>` for `regex`
- adopt CharSequenceAssertionsBuilder, add the new method but keep the one which expects List<String> and delegate to the new one
- adopt the signature of CharSequenceAssertionsImpl, keep only `List<Regex>`

*lib*
- add an additional type parameter `SC` to CharSequenceContains.Searcher which represents the search criterion => use it for `searchFor`
  - use Any as second type parameter in ch.tutteli.atrium.assertions.charsequence.contains.CharSequenceContains.Searcher
  - use Regex for RegexSearcher
  - use String for IndexSearcher, IgnoringCaseIndexSearcher and IgnoringCaseRegexSearcher
  - adopt CharSequenceContainsAssertionCreator.search (use SC)
  - adopt the signature of `_containsRegex`
  - adopt createAssertionGroup and other functions in ch/tutteli/atrium/domain/robstoll/lib/creating/charsequence/contains/creators/creators.kt
    - this means i.a. that we need to map `expected` from Any to String with `toString()` for _containsValues and _containsValuesIgnoringCase

*api*
- provide an additional overload for `containsRegex` in charSequenceAssertions which expects Regex instead of string (see existing fun)
- provide an additional overload for `CharSequenceContains.CheckerOption<T, NoOpSearchBehaviour>.regex` in charSequenceContainsCreators.kt which expects Regex
- add a RegexSpec to ch.tutteli.atrium.api.fluent.en_GB.CharSequenceContainsRegexAssertionsSpec (similar to StringSpec)

## Your first contribution?
- Write a comment `I'll work on this` if you would like to take this issue over.
  This way we get the chance to revise the description in case things have changed in the meantime,
  we might give you additional hints and we can assign the task to you, so that others do not start as well.
- See [Your first code contribution](https://github.com/robstoll/atrium/blob/master/.github/CONTRIBUTING.md#your-first-code-contribution) for guidelines.
- Do not hesitate to ask questions here or to contact us via [Atrium's slack channel](https://kotlinlang.slack.com/team/U3DE1TXKP) if you need help
  ([Invite yourself](https://slack.kotlinlang.org/) in case you do not have an account yet).

**Comment by neelkamath** [comment]
I'll work on this

**Comment by robstoll** [comment]
Implemented with 1beabd3
